shelley/chain-and-ledger/executable-spec/test/Rules/TestUtxo.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Rules.TestUtxo where
+
+import           Data.Word (Word64)
+
+import           Hedgehog (Property, forAll, property, withTests)
+
+import           Control.State.Transition.Generator (ofLengthAtLeast, trace)
+import           Control.State.Transition.Trace (pattern SourceSignalTarget, signal, source,
+                     sourceSignalTargets, target)
+
+import           Coin (pattern Coin)
+import           TxData (pattern Tx, _wdrls)
+import           UTxO (balance)
+
+import           LedgerState (pattern UTxOState)
+import           MockTypes (UTXO)
+
+import           Test.Utils (assertAll)
+
+------------------------------
+-- Constants for Properties --
+------------------------------
+
+numberOfTests :: Word64
+numberOfTests = 300
+
+traceLen :: Word64
+traceLen = 100
+
+--------------------------
+-- Properties for UTXOW --
+--------------------------
+
+-- | Property that checks that the fees are non-decreasing
+feesNonDecreasing :: Property
+feesNonDecreasing = withTests (fromIntegral numberOfTests) . property $ do
+  tr <- fmap sourceSignalTargets $ forAll (trace @UTXO traceLen `ofLengthAtLeast` 1)
```

Comment:
Oh, BTW, in the other PR you used a let bind for the `sourceSignalTargets`. Maybe we want to change that one to match this.